crates/engine/src/prepare/operation/complexity_control.rs

```diff
@@ -135,48 +147,53 @@ fn calculate_child_count<'a>(
     field: OperationWalker<'_, crate::operation::BoundFieldId>,
     list_size_directive: Option<schema::ListSizeDirective<'a>>,
     preset_list_size: Option<usize>,
-) -> ListSizeHandling<'a> {
+) -> PrepareResult<ListSizeHandling<'a>> {
     let field_is_list = field
         .definition()
         .map(|def| def.ty().wrapping.is_list())
         .unwrap_or_default();
 
-    if !field_is_list {
-        return ListSizeHandling::ThisFieldIsTheList(1);
-    }
-
     if let Some(size) = preset_list_size {
-        return ListSizeHandling::ThisFieldIsTheList(size);
+        return Ok(ListSizeHandling::ThisFieldIsTheList(size));
     }
 
+    let default_multiplier = if field_is_list { context.default_list_size } else { 1 };
+
     let Some(directive) = list_size_directive else {
-        return ListSizeHandling::ThisFieldIsTheList(context.default_list_size);
+        return Ok(ListSizeHandling::ThisFieldIsTheList(default_multiplier));
     };
 
     let mut multiplier = directive.assumed_size.unwrap_or(context.default_list_size as u32) as usize;
 
     let mut slicing_arguments = directive.slicing_arguments().peekable();
     if slicing_arguments.peek().is_some() {
-        let argument_size = slicing_arguments
+        let slicing_arguments = slicing_arguments
             .filter_map(|argument| field.argument(argument.name())?.value(context.variables).as_usize())
-            .max();
+            .collect::<Vec<_>>();
+
+        if directive.require_one_slicing_argument && slicing_arguments.len() != 1 {
```

Comment:
shouldn't that be `!slicing_arguments.is_empty()`? or is two arguments also an error?

Reply:
Ok. well might be worth a comment imho and/or rename the error to `ExactlyOne` rather than `One`. Up to you 🤷